hail/src/main/scala/is/hail/expr/ir/Analyses.scala

```diff
@@ -0,0 +1,8 @@
+package is.hail.expr.ir
+
+import is.hail.backend.ExecuteContext
+
+case class Analyses(ir: BaseIR, ctx:ExecuteContext) {
+  val requirednessAnalysis = Requiredness(ir, ctx)
+  val distinctKeyedAnalysis = DistinctlyKeyed.apply(ir)
+}
```

Comment:
This is a weird thing to do. case classes are supposed to be sort of like structs, they hold some basic data. It would make more sense to do like:

```
object Analyses {
    def apply(ir: BaseIR, ctx: ExecuteContext): Analyses = {
         val requirednessAnalysis = Requiredness(ir, ctx)
         val distinctKeyedAnalysis = DistinctlyKeyed.apply(ir)
         Analyses(requirednessAnalysis, distinctKeyedAnalysis)
    }
}

case class Analyses(req: RequirednessAnalysis, dist: DistinctnessAnalysis)